projects/packages/search/src/class-plan.php

```diff
@@ -16,7 +16,27 @@
  * Registers the REST routes for Search.
  */
 class Plan {
-	const JETPACK_SEARCH_PLAN_INFO_OPTION_KEY = 'jetpack_search_plan_info';
+	const JETPACK_SEARCH_PLAN_INFO_OPTION_KEY  = 'jetpack_search_plan_info';
+	const JETPACK_SEARCH_EVER_SUPPORTED_SEARCH = 'jetpack_search_ever_supported_search';
+
+	/**
+	 * Whether we have hooked the actions.
+	 *
+	 * @var boolean
+	 */
+	protected static $hooks_inited = false;
+
+	/**
+	 * Init hooks for updating plan info
+	 */
+	public function init_hooks() {
+		// Update plan info from WPCOM on Jetpack heartbeat.
+		// TODO: implement heartbeart for search.
+		if ( ! static::$hooks_inited ) {
+			add_action( 'jetpack_heartbeat', array( $this, 'get_plan_info_from_wpcom' ) );
+			static::$hooks_inited = true;
+		}
```

Comment:
```suggestion
		if ( ! static::$heartbeat_initialized ) {
			add_action( 'jetpack_heartbeat', array( $this, 'get_plan_info_from_wpcom' ) );
			static::$heartbeat_initialized = true;
		}
```

I recommend using a more descriptive variable name like above here.

Reply:
Changed to `$update_plan_hook_initialized` in eaf94573